src/Step/Deploy/PreDeploy/CleanRedisCache.php

```diff
@@ -74,12 +75,15 @@ public function execute()
             }
             $redisConfig = $cacheConfig['backend_options'];
             $this->logger->info("Clearing redis cache: $cacheType");
-            $cmd = 'redis-cli';
-            $cmd .= isset($redisConfig['server']) ? ' -h ' . $redisConfig['server'] : '';
-            $cmd .= isset($redisConfig['port']) ? ' -p ' . $redisConfig['port'] : '';
-            $cmd .= isset($redisConfig['database']) ? ' -n ' . $redisConfig['database'] : '';
-            $cmd .= ' flushdb';
-            $this->shell->execute($cmd);
+            $redisClient = new Credis_Client(
+                isset($redisConfig['server']) ? $redisConfig['server'] : '127.0.0.1',
```

Comment:
Please use `??` ternary operator.
`$redisConfig['sever'] ?? '127.0.0.1'`

Reply:
I would suggest wrapping this into simple Factory class, to avoid using `new` operator inside of a step class

Reply:
@shiftedreality do you have a suggestion in which namespace I should put the `CredisClientFactory`? Would `Magento\MagentoCloud\Util`be a good place?

Reply:
Yes, that's fine